src/logic/contracts/safeContracts.ts

```diff
@@ -12,13 +12,16 @@ import { calculateGasOf, calculateGasPrice } from 'src/logic/wallets/ethTransact
 import { getWeb3, getNetworkIdFrom } from 'src/logic/wallets/getWeb3'
 import { GnosisSafe } from 'src/types/contracts/GnosisSafe.d'
 import { GnosisSafeProxyFactory } from 'src/types/contracts/GnosisSafeProxyFactory.d'
+import { IProxy } from 'src/types/contracts/IProxy'
+import { getValidMasterCopies } from './api/fetchValidMasterCopies'
 
 export const SENTINEL_ADDRESS = '0x0000000000000000000000000000000000000001'
 export const MULTI_SEND_ADDRESS = '0x8d29be29923b68abfdd21e541b9374737b49cdad'
 export const SAFE_MASTER_COPY_ADDRESS = '0x34CfAC646f301356fAa8B21e94227e3583Fe3F5F'
 export const DEFAULT_FALLBACK_HANDLER_ADDRESS = '0xd5D82B6aDDc9027B22dCA772Aa68D5d74cdBdF44'
 export const SAFE_MASTER_COPY_ADDRESS_V10 = '0xb6029EA3B2c51D09a50B53CA8012FeEB05bDa35A'
 
+const web3 = getWeb3()
```

Comment:
I'm pretty sure that doing this may lead to potential errors. I think it was ok by calling `getWeb3` from within every function.

If the provider changes during runtime, this pointer won't get updated.

Also, by _memoizing_ you're doing the same thing, you're getting an instance of a Contract instantiated with a particular provider. If it changes during runtime, you'll try to _send_ a transaction, and it will fail.


Please, correct me if I'm wrong.


Also, if you're just using a contract instance only to _call_ a method, it's better to use the `web3ReadOnly` exported ref.